cmd/api/routers/routers.go

```diff
@@ -31,171 +31,64 @@ type Controller struct {
 
 const listString = `{"kind": "List", "apiVersion": "v1", "metadata": {"continue": "%s"}, "items": [%s]}`
 
-// GetAllResources responds with the list of resources of a specific type across all namespaces
-func (c *Controller) GetAllResources(context *gin.Context) {
-	group := context.Param("group")
-	version := context.Param("version")
-	apiVersion := fmt.Sprintf("%s/%s", group, version)
+func (c *Controller) GetResources(context *gin.Context) {
 	limit, id, date := pagination.GetValuesFromContext(context)
-
 	kind, err := discovery.GetAPIResourceKind(context)
 	if err != nil {
 		abort.Abort(context, err, http.StatusInternalServerError)
 		return
 	}
 
-	resources, lastId, lastDate, err := c.Database.QueryResources(context.Request.Context(), kind, apiVersion, limit, id, date)
-	if err != nil {
-		abort.Abort(context, err, http.StatusInternalServerError)
-		return
-	}
-
-	continueToken := pagination.CreateToken(lastId, lastDate)
-	context.String(http.StatusOK, listString, continueToken, strings.Join(resources, ","))
-}
-
-func (c *Controller) GetNamespacedResources(context *gin.Context) {
-	group := context.Param("group")
-	version := context.Param("version")
-	apiVersion := fmt.Sprintf("%s/%s", group, version)
-	namespace := context.Param("namespace")
-	limit, id, date := pagination.GetValuesFromContext(context)
-
-	kind, err := discovery.GetAPIResourceKind(context)
-	if err != nil {
-		abort.Abort(context, err, http.StatusInternalServerError)
-		return
-	}
-
-	resources, lastId, lastDate, err := c.Database.QueryNamespacedResources(context.Request.Context(), kind, apiVersion, namespace, limit, id, date)
-	if err != nil {
-		abort.Abort(context, err, http.StatusInternalServerError)
-		return
-	}
-
-	continueToken := pagination.CreateToken(lastId, lastDate)
-	context.String(http.StatusOK, listString, continueToken, strings.Join(resources, ","))
-}
-
-func (c *Controller) GetNamespacedResourceByName(context *gin.Context) {
 	group := context.Param("group")
 	version := context.Param("version")
-	apiVersion := fmt.Sprintf("%s/%s", group, version)
 	namespace := context.Param("namespace")
 	name := context.Param("name")
 
-	kind, err := discovery.GetAPIResourceKind(context)
-	if err != nil {
-		abort.Abort(context, err, http.StatusInternalServerError)
-		return
+	apiVersion := version
+	if group != "" {
+		apiVersion = fmt.Sprintf("%s/%s", group, version)
 	}
 
-	resource, err := c.Database.QueryNamespacedResourceByName(context.Request.Context(), kind, apiVersion, namespace, name)
-	if err != nil {
-		abort.Abort(context, err, http.StatusInternalServerError)
-		return
-	}
+	// We send namespace even if its an empty string (non-namespaced resources) the Database
+	// knows what to do
+	resources, lastId, lastDate, err := c.Database.QueryResources(context.Request.Context(), kind, apiVersion, namespace, name, limit, id, date)
```

Comment:
We should be mindful with long lines,  line break suggestion and also added a couple of fixes in the comment
```suggestion
	// We send name and namespace even if it's an empty string (non-namespaced resources) the Database
	// knows what to do
	resources, lastId, lastDate, err := c.Database.QueryResources(context.Request.Context(), 
	    kind, apiVersion, namespace, name, limit, id, date)
```

Reply:
Done